span.go

```diff
@@ -356,9 +356,17 @@ func (s *Span) End() {
 			}
 		}
 	}
-	if !s.dropped() && len(s.stacktrace) == 0 &&
-		s.Duration >= s.stackFramesMinDuration {
+	switch s.stackFramesMinDuration {
+	case -1:
+		// Always set stacktrace
 		s.setStacktrace(1)
+	case 0:
+		// If s.stackFramesMinDuration == 0, we never set stacktrace.
+	default:
+		if !s.dropped() && len(s.stacktrace) == 0 &&
+			s.Duration >= s.stackFramesMinDuration {
+			s.setStacktrace(1)
+		}
```

Comment:
This could have been squished into a single conditional, and I'm happy to do so. This just seemed to be more easy to read.

Reply:
Looks good to me. Maybe we can revisit when we get to https://github.com/elastic/apm-agent-go/issues/1142